components/FourYearPlanPage.tsx

```diff
@@ -114,20 +61,36 @@ export const FourYearPlanPage: FC<FourYearPlanType> = memo(
 
     // Stuff for category dropdown.
     const [categories, setCategories] = useState<string[]>([]); // list of all categories
-    const [coursesInCategory, setcoursesInCategory] = useState<CourseType[]>([]); // courses in category that is selected
+    const [coursesInCategory, setCoursesInCategory] = useState<CourseType[]>([]); // courses in category that is selected
 
     // Used to keep track of which information to display in the far right area
     const defaultInformationType = "Requirements (Calculated)"; // The default
-    const [informationTypes, setInformationTypes] = useState<string[]>([
-      defaultInformationType
-    ]);
-    const [displayedInformationType, setDisplayedInformationType] =
-      useState<string | undefined>(defaultInformationType);
+    const [informationTypes, setInformationTypes] = useState<string[]>([defaultInformationType]);
+    const [displayedInformationType, setDisplayedInformationType] = useState<string | undefined>(defaultInformationType);
+
+    function initializeSemesters(): SemesterType[] {
+      const tempSemesters = [];
+      for (let i = 0; i < semestersLength; i++) {
+        tempSemesters.push(
+          {
+            accepts: [ItemTypes.COURSE],
+            courses: [],
+            semesterNumber: i + 1,
+            SemesterCredits: 0,
+            Warning: ""
+          }
+        );
+      }
+      return tempSemesters;
+    }
 
     useEffect(() => {
       // Whenever completed courses may update, determine
       // whether we need to display it in the dropdown
-      if (userMajor()?.completed_courses !== undefined) {
+      let completedCourses = userMajor()?.completed_courses !== undefined ? userMajor()?.completed_courses.length : 0;
```

Comment:
This can be shorter using the `??` operator. It should look like `const completedCourses = userMajor()?.completed_courses.length ?? 0;`